streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala

```diff
@@ -374,26 +374,53 @@ abstract class DStream[T: ClassTag] (
    * Wrap a body of code such that the call site and operation scope
    * information are passed to the RDDs created in this body properly.
    */
-  protected def createRDDWithLocalProperties[U](time: Time)(body: => U): U = {
+  protected[streaming] def createRDDWithLocalProperties[U](time: Time)(body: => U): U = {
+    createRDDWithLocalProperties(time, makeRDDOpsVisibleInUI = false)(body)
+  }
+
+  /**
+   * Wrap a body of code such that the call site and operation scope
+   * information are passed to the RDDs created in this body properly.
+   */
+  protected[streaming] def createRDDWithLocalProperties[U](
+      time: Time,
+      makeRDDOpsVisibleInUI: Boolean)(body: => U): U = {
     val scopeKey = SparkContext.RDD_SCOPE_KEY
     val scopeNoOverrideKey = SparkContext.RDD_SCOPE_NO_OVERRIDE_KEY
     // Pass this DStream's operation scope and creation site information to RDDs through
     // thread-local properties in our SparkContext. Since this method may be called from another
     // DStream, we need to temporarily store any old scope and creation site information to
     // restore them later after setting our own.
-    val prevCallSite = ssc.sparkContext.getCallSite()
+    val prevCallSite = CallSite(
+      ssc.sparkContext.getLocalProperty(CallSite.SHORT_FORM),
+      ssc.sparkContext.getLocalProperty(CallSite.LONG_FORM)
+    )
     val prevScope = ssc.sparkContext.getLocalProperty(scopeKey)
     val prevScopeNoOverride = ssc.sparkContext.getLocalProperty(scopeNoOverrideKey)
 
     try {
-      ssc.sparkContext.setCallSite(creationSite)
+      if (makeRDDOpsVisibleInUI) {
+        // Unset the short form call site, so that generated RDDs get their own short form callsite
+        ssc.sparkContext.setLocalProperty(CallSite.SHORT_FORM, null)
+      } else {
+        ssc.sparkContext.setLocalProperty(CallSite.SHORT_FORM, creationSite.shortForm)
+      }
+
+      // Set the long form call site so that the stack trace shows location of DStream operation
+      // rather than the internals of Spark Streaming where RDD was generated
+      ssc.sparkContext.setLocalProperty(CallSite.LONG_FORM, creationSite.longForm)
```

Comment:
why do you want to show the long form regardless of whether `makeRDDOpsVisibleInUI` is true?